### PR TITLE
support for vios to copy list and delete vios images

### PR DESCRIPTION
**Pull request by chiranthanmv**

### CODE REVIEW

File: playbooks/demo_create_and_install_vios.yaml [hunk 1]
```diff
@@ -14,6 +14,41 @@
               6365
 
   tasks:
```
Comment (naveenkandakur) [inline playbooks/demo_create_and_install_vios.yaml]: Don't Add this part in the playbook, may be once Manya or Sreenidhi's code is ready you can create one playbook to copy the image and install or upgarde vios

File: plugins/module_utils/hmc_resource.py [hunk 2]
```diff
@@ -805,3 +805,31 @@ def list_all_lpars_details(self, sys_name, filter=None):
     def getSystemNameFromMTMS(self, system_name):
         attr_dict = self.getManagedSystemDetails(system_name)
         return attr_dict.get('name')
+
+    def copyViosImage(self, sftp_server, sftp_user, sftp_password, image_name):
```
Comment (naveenkandakur) [inline plugins/module_utils/hmc_resource.py]: this works only for SFTP? this support is not available for FTP or NFS?
If support is available then we have to make the code changes accordingly

File: plugins/module_utils/hmc_resource.py [hunk 3]
```diff
@@ -805,3 +805,31 @@ def list_all_lpars_details(self, sys_name, filter=None):
...
+
+        self.hmcconn.execute(cpviosimgCmd)
+
+    def listViosImages(self, image_name=None):
+        if image_name:
+            lsviosimgCmd = self.CMD['LSVIOSIMG'] +\
+                '| grep -w ' +  image_name +\
+                ' || echo No results were found'
+        else:
+            lsviosimgCmd = self.CMD['LSVIOSIMG']
```
Comment (naveenkandakur) [inline plugins/module_utils/hmc_resource.py:860]: if there are no images exist then this Line of code also returns as "No results were found" ?

File: plugins/module_utils/hmc_resource.py [hunk 4]
```diff
@@ -805,3 +805,31 @@ def list_all_lpars_details(self, sys_name, filter=None):
...
+        output = self.hmcconn.execute(lsviosimgCmd)
+        if 'No results were found' in output:
+            return []
+        return self.cmdClass.parseMultiLineCSV(output)
+
+    def deleteViosImage(self, image_name):
+        rmviosimgCmd = self.CMD['RMVIOSIMG'] +\
+            self.OPT['RMVIOSIMG']['-N'] + image_name
+        self.hmcconn.execute(rmviosimgCmd)
```
Comment (naveenkandakur) [inline plugins/module_utils/hmc_resource.py]: what happens if we try to delete the image which doesn't exist?

**Review by naveenkandakur — CHANGES_REQUESTED** [review]
Parameter checks need to be handled, cpvios if supports NFS and FTP that supporting code needs to be added
Unit test needs to be added and documentation needs to be added

File: plugins/module_utils/hmc_resource.py (hunk 2 above)
Comment (chiranthanmv) [reply plugins/module_utils/hmc_resource.py]: Added the support for NFS.

File: plugins/module_utils/hmc_resource.py (hunk 3 above)
Comment (chiranthanmv) [reply plugins/module_utils/hmc_resource.py:860]: yes the cmd will return "No results were found" string

File: plugins/module_utils/hmc_resource.py (hunk 4 above)
Comment (chiranthanmv) [reply plugins/module_utils/hmc_resource.py]: it returns "HSCLC464 Specified VIOS image test was not found." and this is handled in https://github.com/chiranthanmv/ansible-power-hmc/pull/1/files#diff-2f46ceb7b51fc324580e8abc826831ab4504887cac63b0902e25dedc51c0fc4dR673

**Comment by chiranthanmv** [comment]
> Parameter checks need to be handled, cpvios if supports NFS and FTP that supporting code needs to be added Unit test needs to be added and documentation needs to be added

Added support for Param checks and NFS, FTP is not supported for cmd 'cpviosimg', Unit test has been added for copy action and also have added the documentation

File: playbooks/demo_create_and_install_vios.yaml (hunk 1 above)
Comment (chiranthanmv) [reply playbooks/demo_create_and_install_vios.yaml]: Will create a playbook once install support us added

**Pull request merged**